docs/messaging.rst

```diff
@@ -1122,6 +1122,41 @@ Frontends can choose how they update prior outputs (or if they regard this as a
 regular ``display_data`` message). Within the jupyter and nteract_ notebooks,
 all displays that match the ``display_id`` are updated (even if there are multiple).
 
+Transient Display Data
+----------------------
+
+.. versionadded:: 5.3
+
+This message is very similar to `display_data` but its is content is meant to be
+transient (e.g. not saved in output areas of notebooks). Contents that can be sent
+through this message type can include but not limited to status information of long
+calculation and debug information.
+
+.. _transient_display_data:
+
+Message type: ``transient_display_data``::
+
+    content = {
+
+        # Description of the data
+        'title': string,
+
+        # The data dict contains key/value pairs, where the keys are MIME
+        # types and the values are the raw data of the representation in that
+        # format.
+        'data' : dict,
+
+        # A boolean flag, which, if True, will not clear existing displayed
+        # message with the same title.
+        'append': False,
+    }
+
+
+Frontends can choose where and how to display and update `transient_display_data`.
+Due to the transient nature of such messages, a `transient_display_data` message
+would generally replace prior output although multiple messages with the same
```

Comment:
Would it make sense if we used `update_display_data` to target a `transient_display_data` that was already on the screen? Then you'd able to update an individual output and maintain consistency amongst the other `display_data` like messages.

I'm assuming you'd want to have a targeted `clear_output` message too then to clear all of them.

Reply:
It does not seem to be a good idea to multi-purpose `update_display_data` here, especially because there is no `display_id` for `transient_display_data`.

How exactly the messages should be aggregated (appended) and cleared depends on user interaction. I think the use pattern is generally something like this:

1. User does something (e.g. executes a cell)
2. Transient messages appear, and probably aggregate
3. User consumes the message and performs the next action (e.g. modify the code and re-execute).
4. A new batch of transient messages replaces displayed one.

So in my design:
1. `title` shows the intent of one or more messages.
2. `append=true` would append message to a previous message with the same `title`

In practice:
1. If a kernel wants to display some debug or progress information during the execution, it can send multiple `transient_display_data` with the same `title` and `append=True`.
2. If a kernel would like to update previous message (e.g. new CPU/MEM status), it can either send a message with a different title, or the same title with `append=False`.